actions/AuthAction.js

```diff
@@ -142,44 +140,34 @@ async function checkForEarlyProductAccess(toggle, sdkKey, isInternal, org) {
   });
 }
 
-function asAuthAction(action) {
+function asAuthAction(action, isAuthorize = true) {
   return async (params) => {
     const imsEndpoint = params.IMS_ENDPOINT;
     const clientId = params.IMS_CLIENT_ID;
-    const serviceClientId = params.IMS_SERVICE_CLIENT_ID;
-    const clientSecret = params.IMS_SERVICE_CLIENT_SECRET;
-    const permAuthCode = params.IMS_SERVICE_PERM_AUTH_CODE;
     const productContext = params.IMS_PRODUCT_CONTEXT;
     const earlyAccessToggle = params.FT_EARLY_ACCESS;
     const ldSdkKey = params.LD_SDK_KEY;
 
-    const accessToken = getAccessToken(params);
     const imsOrg = getImsOrg(params);
+    const accessToken = await getAccessToken(params);
 
-    // Validate the access token
-    if (!await isValidToken(imsEndpoint, clientId, accessToken)) {
-      throw new Error('Access token is invalid');
-    }
-
-    // Check that the profile has access to the product
-    const imsProfile = await getImsProfile(imsEndpoint, clientId, accessToken);
-    if (!imsProfile) {
-      throw new Error('Failed to fetch profile');
-    }
+    if (isAuthorize) {
+      // Check that the profile has access to the product
+      const imsProfile = await getImsProfile(imsEndpoint, clientId, accessToken);
+      if (!imsProfile) {
+        throw new Error('Failed to fetch profile');
+      }
 
-    if (!await checkForProductContext(imsProfile, imsOrg, productContext)) {
-      const isInternalUser = checkForAdobeInternalUser(imsProfile);
+      if (!await checkForProductContext(imsProfile, imsOrg, productContext)) {
+        const isInternalUser = checkForAdobeInternalUser(imsProfile);
 
-      if (!await checkForEarlyProductAccess(earlyAccessToggle, ldSdkKey, isInternalUser, imsOrg)) {
-        throw new Error('Profile does not have access to the product');
+        if (!await checkForEarlyProductAccess(earlyAccessToggle, ldSdkKey, isInternalUser, imsOrg)) {
+          throw new Error('Profile does not have access to the product');
+        }
       }
     }
 
-    // If everything is okay, generate a service token
```

Comment:
A service token is needed only for actions communicating with Firefall, so I created a new `asFirefallAction` decorator for that purpose.